modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -1049,8 +1044,23 @@ public void init(
     }
 
     /**
-     * Recovers components state on start by invoking configuration listeners ({@link #notifyConfigurationListeners()}
-     * and deploying watches after that.
+     * Initializes the cluster configuration with the specified user-provided configuration upon cluster initialization.
+     */
+    private CompletableFuture<Void> initializeClusterConfiguration(ExecutorService startupExecutor) {
+        return cmgMgr.initialClusterConfigurationFuture().thenAcceptAsync(cfg -> {
+            if (cfg == null) {
+                return;
+            }
```

Comment:
I would recommend using EMPTY_CFG_SOURCE (or whatever that name is), it would simplify assertions in configuration changer.